src/index.tsx

```diff
@@ -143,6 +149,28 @@ function FastImageBase({
     ...props
 }: FastImageProps & { forwardedRef: React.Ref<any> }) {
     const resolvedSource = Image.resolveAssetSource(source as any)
+    let resolvedDefaultSource:
+        | ImageRequireSource
+        | string
+        | null = defaultSource
+
+    if (isAndroid) {
```

Comment:
```suggestion
        | typeof defaultSource
        | string
        | null = defaultSource

    if (isAndroid && defaultSource) {
```
This fixes typescript build errors

Reply:
There was a `&& defaultSource` there but TS complained. Did you test these changes?

Reply:
The check should stay `if (isAndroid) else ...`. First case is for Android, `else` is for iOS. In iOS they are represented as numbers.

Reply:
The problem is `defaultSource` is optional, its value can be `undefined`. But `Image.resolveAssetSource` input parameter cannot be `undefined`.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-native/index.d.ts
```
resolveAssetSource(source: ImageSourcePropType)
export type ImageSourcePropType = ImageURISource | ImageURISource[] | ImageRequireSource;
export type ImageRequireSource = number;
```

Therefore I added `&& defaultSource` to if condition. As an alternative we can also do like this - 
```
if (!defaultSource) {
    resolvedDefaultSource = null
} else if (isAndroid) {
    ....
} else if (typeof resolvedDefaultSource !== 'number') {
    ....
}
```